fcrepo/fcrepo
# Parallelize index rebuilding

**JIRA Ticket**: https://jira.lyrasis.org/browse/FCREPO-3503

# What does this Pull Request do?
Splits the reindexing of OCFL objects into individual tasks which can be shared amongst threads to speed up reindexing.

This is all still done in a single "transaction" for each service which is then committed at the end.

The unit tests for the manager and service might be too weak so I am open to adding more.

# How should this be tested?

Create a repo with a bunch of objects, remove the ocflIndex database. Restart Fedora and see how long it takes to finish reindexing with and before this PR.

# Interested parties
@fcrepo/committers
